modules/auth/register-service.spec.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @jest-environment node
+ */
+
+import { supabase } from "@/services/auth/auth-supabase.service";
+import { registerUser } from "./register-service";
+
+describe("registerService (Integration Test)", () => {
+  const validEmail = "test@example.com";
+  const validPassword = "strong_password";
+
+  beforeEach(async () => {
+    const { data, error } = await supabase
```

Comment:
Im concern to expose supabase in the test cases, as is a third party. 

Here we just want to check the frontend not the backend (backend will check him self, maybe is superbase maybe is other thing in the future)

For that we need to mock up the service layer. 

We put the data click, calls the API.
Apu return success -> redirect
Api returns needs to validate -> go right page
APi returns error -> display the error

Is what we care here fir